### PR TITLE
org.flightgear.FlightGear

### PR DESCRIPTION
**Pull request by Alexander-Wilms**
Additional aircraft & scenery can be downloaded from flightgear.org and must be placed in ~/.fgfs/Aircraft, respective ~/.fgfs/Scenery.

I'm using the KDE Sdk, since FlightGear offers a Qt GUI, which I've disabled for now. The FlightGear code makes an assertion in the Qt code fail, but the root cause itself is not fatal. But since the Qt contained in the KDE Platform is build with the -debug flag, this causes a crash.

Upstream bug report: https://sourceforge.net/p/flightgear/codetickets/2012/

### CODE REVIEW

File: org.flightgear.FlightGear.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2014 Richard Hughes <richard@hughsie.com> -->
+
+<!-- HOUSEKEEPING, REMOVE THIS COMMENT WHEN THIS GOES UPSTREAM
+BugReportURL: https://sourceforge.net/p/fgrun/support-requests/9/
+SentUpstream: 2014-09-17
+-->
+
+<application>
+  <id type="desktop">org.flightgear.FlightGear</id>
```
Comment (TingPing) [inline org.flightgear.FlightGear.appdata.xml]: Seems to be an odd mix between old and new format. Run `appstream-util upgrade` on it perhaps.

File: org.flightgear.FlightGear.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  </description>
+  <url type="homepage">http://www.flightgear.org/</url>
+  <screenshots>
+    <screenshot type="default">http://wiki.flightgear.org/images/b/b3/SOTM-Feb18.jpg</screenshot>
+    <screenshot>http://wiki.flightgear.org/images/3/37/SOTY-2017.jpeg</screenshot>
+    <screenshot>http://wiki.flightgear.org/images/2/27/SOTM-Oct17.png</screenshot>
+  </screenshots>
+  <!-- FIXME: change this to an upstream email address for spec updates
+  <updatecontact>someone_who_cares@upstream_project.org</updatecontact>
+   -->
```
Comment (TingPing) [inline org.flightgear.FlightGear.appdata.xml]: Add `releases`

File: org.flightgear.FlightGear.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+<!-- Copyright 2014 Richard Hughes <richard@hughsie.com> -->
+
+<!-- HOUSEKEEPING, REMOVE THIS COMMENT WHEN THIS GOES UPSTREAM
+BugReportURL: https://sourceforge.net/p/fgrun/support-requests/9/
+SentUpstream: 2014-09-17
+-->
+
+<application>
+  <id type="desktop">org.flightgear.FlightGear</id>
+  <metadata_license>CC0-1.0</metadata_license>
```
Comment (TingPing) [inline org.flightgear.FlightGear.appdata.xml]: Missing `<launchable>`

File: org.flightgear.FlightGear.json [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	"command":"flightgear.sh",
+	"rename-icon":"flightgear",
+	"finish-args":[
+		"--share=ipc",
+		"--socket=x11",
+		"--socket=wayland",
+		"--device=dri",
+		"--socket=pulseaudio",
+		"--share=network",
+		"--filesystem=~/.fgfs/"
```
Comment (TingPing) [inline org.flightgear.FlightGear.json]: `--persist=.fgfs`

File: org.flightgear.FlightGear.json [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+	"id":"org.flightgear.FlightGear",
+	"runtime":"org.kde.Platform",
+	"runtime-version":"5.10",
+	"sdk":"org.kde.Sdk",
+	"command":"flightgear.sh",
+	"rename-icon":"flightgear",
+	"finish-args":[
+		"--share=ipc",
+		"--socket=x11",
+		"--socket=wayland",
```
Comment (TingPing) [inline org.flightgear.FlightGear.json]: Does it actually support wayland?

File: org.flightgear.FlightGear.json [hunk 6]
```diff
@@ -0,0 +1,193 @@
+{
+	"id":"org.flightgear.FlightGear",
```
Comment (TingPing) [inline org.flightgear.FlightGear.json]: nitpick but single space between keys and values please.

File: org.flightgear.FlightGear.json [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+				"--disable-postproc",
+
+				/* Enough to run the base game, might need extending for mod content */
+				"--enable-decoder=bink,binkaudio_rdft,mp3,pcm_s16le,vorbis",
+				"--enable-demuxer=bink,mp3,ogg,pcm_s16le,wav"
+			],
+			"sources": [
+				{
+				"type": "archive",
+				"url": "http://ffmpeg.org/releases/ffmpeg-2.8.14.tar.xz",
```
Comment (TingPing) [inline org.flightgear.FlightGear.json]: Very old, does it not support 3.4?

File: flightgear.sh [hunk 8]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+FG_ROOT=/app/share/flightgear LD_LIBRARY_PATH=/app/lib64 FG_AIRCRAFT=~/.fgfs/Aircraft/ FG_SCENERY=~/.fgfs/Scenery fgfs "$@"
```
Comment (TingPing) [inline flightgear.sh]: All libs should go into `/app/lib` and the `LD_LIBRARY_PATH` var can go.

File: org.flightgear.FlightGear.json (hunk 5 above)
Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.json]: Apparently not sufficiently: https://forum.flightgear.org/viewtopic.php?f=37&t=32169

File: org.flightgear.FlightGear.json (hunk 4 above)
Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.json]: This is needed so one can use additional aircraft. By default, there's only one shipped with the Flatpak.

File: org.flightgear.FlightGear.appdata.xml (hunk 3 above)
Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.appdata.xml]: Will do

File: org.flightgear.FlightGear.appdata.xml (hunk 2 above)
Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.appdata.xml]: Will do

File: org.flightgear.FlightGear.json (hunk 6 above)
Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.json]: Will do

File: org.flightgear.FlightGear.json (hunk 7 above)
Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.json]: It doesn't recognize it. I chose 2.8.z due to this forum entry: http://forum.openscenegraph.org/viewtopic.php?t=16424

Comment (TingPing) [reply org.flightgear.FlightGear.json]: https://git.archlinux.org/svntogit/community.git/tree/trunk/openscenegraph-ffmpeg3.patch?h=packages/openscenegraph

File: org.flightgear.FlightGear.json (hunk 4 above)
Comment (TingPing) [reply org.flightgear.FlightGear.json]: The directory is still accessible. It is a hidden directory anyway so it requires user documentation no matter what.

File: org.flightgear.FlightGear.json (hunk 7 above)
Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.json]: That causes a compilation error:
```
/run/build/openscenegraph/src/osgPlugins/ffmpeg/FFmpegDecoderAudio.cpp:103:36: error: ‘CODEC_ID_NONE’ was not declared in this scope
         if (m_context->codec_id == CODEC_ID_NONE)
```

File: org.flightgear.FlightGear.appdata.xml (hunk 1 above)
Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.appdata.xml]: Will do

File: org.flightgear.FlightGear.json (hunk 7 above)
Comment (TingPing) [reply org.flightgear.FlightGear.json]: Hmm, fine for now but do ensure that this is at least reported upstream?

**Comment by TingPing** [comment]
bot, build org.flightgear.FlightGear

File: org.flightgear.FlightGear.json (hunk 7 above)
Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.json]: Even though it's a patched, old version? The next version of FlightGear, released in 24 days, will require at least OSG 3.4

**Comment by TingPing** [comment]
bot, build org.flightgear.FlightGear

**Comment by Alexander-Wilms** [comment]
Thanks. Seems like sourceforge has issues again: 
```
Downloading simgearDownloading https://sourceforge.net/projects/flightgear/files/release-2017.3/simgear-2017.3.1.tar.bz2
Failed to download sources: module simgear: Internal Server Error
```

**Comment by Alexander-Wilms** [comment]
Could you trigger another build?

**Comment by nedrichards** [comment]
bot, build org.flightgear.FlightGear

**Comment by nedrichards** [comment]
bot, build org.flightgear.FlightGear

**Comment by Alexander-Wilms** [comment]
All builds succeeded: https://flathub.org/builds/#/builders/2/builds/98

File: org.flightgear.FlightGear.json (hunk 7 above)
Comment (TingPing) [reply org.flightgear.FlightGear.json]: Oh alright.

File: org.flightgear.FlightGear.json [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+					"path": "org.flightgear.FlightGear.appdata.xml"
+				},
+				{
+					"type": "file",
+					"path": "flightgear.sh"
+				}
+			],
+			"build-commands": [
+				"install -Dm644 ../org.flightgear.FlightGear.appdata.xml /app/share/appdata/org.flightgear.FlightGear.appdata.xml",
+				"install -Dm744 ../flightgear.sh /app/bin/flightgear.sh"
```
Comment (TingPing) [inline org.flightgear.FlightGear.json]: You probably need to patch the desktop file to use the wrapper script?

Comment (Alexander-Wilms) [reply org.flightgear.FlightGear.json]: Right, done.

**Comment by TingPing** [comment]
bot, build org.flightgear.FlightGear

**Comment by Alexander-Wilms** [comment]
All builds succeeded: https://flathub.org/builds/#/builders/2/builds/119

**Comment by TingPing** [comment]
Repository has been created: https://github.com/flathub/org.flightgear.FlightGear

**Comment by TingPing** [comment]
Too late now but you forked the wrong branch so its full of junk...

**Pull request closed without merging**